src/BlockTemplatesController.php

```diff
@@ -855,12 +855,36 @@ public function update_template_permalink( $value, $option ) {
 	 * @return string THe actual permalink assigned to the page. May differ from $permalink if it was already taken.
 	 */
 	protected function sync_endpoint_with_page( $page, $page_slug, $permalink ) {
-		$matching_page = get_page_by_path( $permalink );
+		$matching_page = get_page_by_path( $permalink, OBJECT, 'page' );
+
+		/**
+		 * Filters whether to attempt to guess a redirect URL for a 404 request.
+		 *
+		 * Returning a false value from the filter will disable the URL guessing
+		 * and return early without performing a redirect.
+		 *
+		 * @since 5.5.0
```

Comment:
Why does this PHPDoc comment says `@since 5.5.0`?

Reply:
The PHPDoc was required for the hook per listing definitions, I could disable that (we have some cases) but maintained it as I think any documentation raising awareness for WP URL guessing

Reply:
I wasn't referring to the PHPDoc itself, but to this part: `@since 5.5.0`. This comment indicates that this particular section had been introduced in [WooCommerce Blocks 5.5.0](https://github.com/woocommerce/woocommerce-blocks/releases/tag/v5.5.0), while it appears that it's been introduced with WooCommerce Blocks 11.0.0.